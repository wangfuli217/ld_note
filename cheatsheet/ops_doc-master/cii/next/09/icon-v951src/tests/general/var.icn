#SRC: V9GEN

procedure main(a)
   local x, l0, l1, l2, l3, l4, l5, l6, l7, l8, l9
   static y
   variable("a") := 1
   variable("x") := 2
   variable("y") := 3
   if variable("z") then write("oops z")
   &subject := "abcde"
   &pos := 3
   write("&pos &subject: ", &pos, " ", variable("&subject"))
   variable("&subject") := "wxyz"
   write("&pos &subject: ", &pos, " ", variable("&subject"))
   if variable("&clock") then write("oops &clock")
   main2()
   write()

   # test display()
   # some formats should differ from those of image()
   l0 := []
   l1 := [&letters -- 'jJ']
   l2 := ["supercalifragilistic", "expialidocious"]
   l3 := [&phi, &e, &pi]
   l4 := [1,2,3,4]
   l5 := [1,2,3,4,5]
   l6 := [1,2,3,4,5,6]
   l7 := [1,2,3,4,5,6,7]
   l8 := [1,2,3,4,5,6,7,8]
   l9 := [l0, l1, l2, l3, l4, l5, l6, l7, l8]
   recurse(1, 4)
end

record complex(r,i)

procedure main2(a)
   local T, L, L1
   static s
   every write(image(name(main | T | L | s | a)))
   T := table()
   L := list(200)
   L1 := []
   every 1 to 200 do push(L1,1)
   write(image(L[10]))
   write(image(name(&error)))
   write(image(name(T["abc"])))
   T["abc"] := 1
   write(image(name(T["abc"])))
   every write(image(name(L[1 | 2 | 3 | -1 | -10])))
   every write(image(name(L1[1 | 2 | 3 | -1 | -10])))
   write(image(name(complex().r)))
   return
end

procedure recurse(a,b)	# to build up some context for display() 
   local x, y
   x := a + b
   y := a - b
   if a < 5 then
      recurse(a + 1, b + a)
   else {
      write(repl('-',70))
      display(3, &output)
      write(repl('-',70))
      display(, &output)
      write(repl('-',70))
      }
   return
end
