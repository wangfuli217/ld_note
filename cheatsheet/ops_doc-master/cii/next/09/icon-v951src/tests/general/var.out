&pos &subject: 3 abcde
&pos &subject: 1 wxyz
"main"
"T"
"L"
"s"
"a"
&null
"&error"
"T[\"abc\"]"
"T[\"abc\"]"
"L[1]"
"L[2]"
"L[3]"
"L[200]"
"L[191]"
"L[1]"
"L[2]"
"L[3]"
"L[200]"
"L[191]"
"complex.r"

----------------------------------------------------------------------
co-expression_1(1)

recurse local identifiers:
   a = 5
   b = 14
   x = 19
   y = -9
recurse local identifiers:
   a = 4
   b = 10
   x = 14
   y = -6
recurse local identifiers:
   a = 3
   b = 7
   x = 10
   y = -4

global identifiers:
   complex = record constructor complex
   display = function display
   image = function image
   list = function list
   main = procedure main
   main2 = procedure main2
   name = function name
   push = function push
   recurse = procedure recurse
   repl = function repl
   table = function table
   variable = function variable
   write = function write
----------------------------------------------------------------------
co-expression_1(1)

recurse local identifiers:
   a = 5
   b = 14
   x = 19
   y = -9
recurse local identifiers:
   a = 4
   b = 10
   x = 14
   y = -6
recurse local identifiers:
   a = 3
   b = 7
   x = 10
   y = -4
recurse local identifiers:
   a = 2
   b = 5
   x = 7
   y = -3
recurse local identifiers:
   a = 1
   b = 4
   x = 5
   y = -3
main local identifiers:
   a = 1
   x = 2
   l0 = list_4 = []
   l1 = list_5 = ['ABCDEFGHIKLMNOPQ...']
   l2 = list_6 = ["supercalifragili...","expialidocious"]
   l3 = list_7 = [1.618033989,2.718281828,3.141592654]
   l4 = list_8 = [1,2,3,4]
   l5 = list_9 = [1,2,3,4,5]
   l6 = list_10 = [1,2,3,4,5,6]
   l7 = list_11 = [1,2,3,...,5,6,7]
   l8 = list_12 = [1,2,3,...,6,7,8]
   l9 = list_13 = [list_4 = [],list_5(1),list_6(2),...,list_10(6),list_11(7),list_12(8)]
   y = 3

global identifiers:
   complex = record constructor complex
   display = function display
   image = function image
   list = function list
   main = procedure main
   main2 = procedure main2
   name = function name
   push = function push
   recurse = procedure recurse
   repl = function repl
   table = function table
   variable = function variable
   write = function write
----------------------------------------------------------------------
